spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/SkipPatternProviderConfigTest.java

```diff
@@ -97,190 +125,169 @@ public void should_return_empty_when_no_endpoints() {
-		WebEndpointProperties webEndpointProperties = new WebEndpointProperties();
-		webEndpointProperties.setBasePath("/");
-
-		EndpointsSupplier<ExposableWebEndpoint> endpointsSupplier = () -> {
-			ExposableWebEndpoint infoEndpoint = createEndpoint("info");
-			ExposableWebEndpoint healthEndpoint = createEndpoint("health");
-
-			return Arrays.asList(infoEndpoint, healthEndpoint);
-		};
-
-		Optional<Pattern> pattern = new TraceWebAutoConfiguration.ActuatorSkipPatternProviderConfig()
-				.skipPatternForActuatorEndpointsSamePort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
-
-		then(pattern).isNotEmpty();
-		then(pattern.get().pattern()).isEqualTo("/(info|info/.*|health|health/.*)");
+		contextRunner
+			.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+			.withPropertyValues(		
+				"management.endpoint.health.enabled=true",
+				"management.endpoint.info.enabled=true",
+				"management.endpoints.web.exposure.include=info,health",
+				"management.endpoints.web.base-path=/")
+			.run(context -> {
+				SkipPatternProvider skipPatternProvider = context.getBean(SkipPatternProvider.class);
+				Pattern pattern = skipPatternProvider.skipPattern();
+				then(pattern.pattern()).contains("/(health|health/.*|info|info/.*)");
+			});
 	}
-
+	
 	@Test
 	public void should_return_endpoints_with_context_path_and_base_path_set_to_root() {
-		WebEndpointProperties webEndpointProperties = new WebEndpointProperties();
-		webEndpointProperties.setBasePath("/");
-		ServerProperties properties = new ServerProperties();
-		properties.getServlet().setContextPath("foo");
-
-		EndpointsSupplier<ExposableWebEndpoint> endpointsSupplier = () -> {
-			ExposableWebEndpoint infoEndpoint = createEndpoint("info");
-			ExposableWebEndpoint healthEndpoint = createEndpoint("health");
-
-			return Arrays.asList(infoEndpoint, healthEndpoint);
-		};
-
-		Optional<Pattern> pattern = new TraceWebAutoConfiguration.ActuatorSkipPatternProviderConfig()
-				.skipPatternForActuatorEndpointsSamePort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
-
-		then(pattern).isNotEmpty();
-		then(pattern.get().pattern()).isEqualTo("foo/(info|info/.*|health|health/.*)");
+		contextRunner
+			.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+			.withPropertyValues(		
+				"management.endpoint.health.enabled=true",
+				"management.endpoint.info.enabled=true",
+				"management.endpoints.web.exposure.include=info,health",
+				"management.endpoints.web.base-path=/",
+				"server.servlet.context-path=foo")
+			.run(context -> {
+				SkipPatternProvider skipPatternProvider = context.getBean(SkipPatternProvider.class);
+				Pattern pattern = skipPatternProvider.skipPattern();
+				then(pattern.pattern()).contains("foo/(health|health/.*|info|info/.*)");
+			});
 	}
-
+	
 	@Test
-	public void should_return_endpoints_with_actuator_context_path_set_to_root() {
```

Comment:
This test cases is covered by `should_return_endpoints_without_context_path_and_base_path_set_to_root` and `should_return_endpoints_with_context_path_and_base_path_set_to_ot_different_port`